PWGHF/TableProducer/trackIndexSkimCreator.cxx

```diff
@@ -3276,94 +3377,69 @@ struct HfTrackIndexSkimCreatorLfCascades {
 
-              // std::array<float, 3> secondaryVertex3 = {0., 0., 0.};
-              const auto& secondaryVertex3 = df3.getPCACandidate();
+                df3.propagateTracksToVertex();
 
-              // fill table row
-              rowTrackIndexCasc3Prong(casc.globalIndex(),
-                                      trackPion1.globalIndex(),
-                                      trackPion2.globalIndex());
+                std::array<float, 3> pVec1 = {0.};
+                std::array<float, 3> pVec2 = {0.};
+                std::array<float, 3> pVec3 = {0.};
+                df3.getTrack(0).getPxPyPzGlo(pVec1); // take the momentum at the Xic vertex
+                df3.getTrack(1).getPxPyPzGlo(pVec2);
+                df3.getTrack(2).getPxPyPzGlo(pVec3);
 
-              // fill histograms
-              if (fillHistograms) {
-                registry.fill(HIST("hVtx3ProngX"), secondaryVertex3[0]);
-                registry.fill(HIST("hVtx3ProngY"), secondaryVertex3[1]);
-                registry.fill(HIST("hVtx3ProngZ"), secondaryVertex3[2]);
+                // std::array<float, 3> secondaryVertex3 = {0., 0., 0.};
+                auto secondaryVertex3 = df3.getPCACandidate();
 
-                std::array<std::array<float, 3>, 3> arr3Mom = {pVec1, pVec2, pVec3};
-                for (int iDecay3P = 0; iDecay3P < kN3ProngDecays; iDecay3P++) {
-                  auto mass3Prong = RecoDecay::m(arr3Mom, arrMass3Prong[iDecay3P][0]);
-                  switch (iDecay3P) {
-                    case hf_cand_casc_lf_3prong::DecayType::XicplusToXiPiPi:
-                      registry.fill(HIST("hMassXicPlusToXiPiPi"), mass3Prong);
-                      break;
-                  }
+                if (std::abs(casc.mXi() - massXi) < cascadeMassWindow) {
+                  SETBIT(hfFlag, aod::hf_cand_casc_lf::DecayType3Prong::XicplusToXiPiPi);
                 }
-              }
-            }
-          }
 
-          if (df2.process(trackCasc, trackParVarPion1) == 0) {
-            continue;
-          }
-
-          std::array<float, 3> pVec1 = {0.};
-          std::array<float, 3> pVec2 = {0.};
+                std::array<std::array<float, 3>, 3> arr3Mom = {pVec1, pVec2, pVec3};
+                auto mass3Prong = RecoDecay::m(arr3Mom, arrMass3Prong[hf_cand_casc_lf::DecayType3Prong::XicplusToXiPiPi]);
 
-          df2.getTrack(0).getPxPyPzGlo(pVec1);
-          df2.getTrack(1).getPxPyPzGlo(pVec2);
+                if (mass3Prong < massXiPiPiMin || mass3Prong > massXiPiPiMax) {
+                  statusMassXiPiPiChannel = false;
+                }
 
-          const auto& secondaryVertex2 = df2.getPCACandidate();
+                // fill histograms
+                if (fillHistograms && statusMassXiPiPiChannel) {
+                  registry.fill(HIST("hVtx3ProngX"), secondaryVertex3[0]);
+                  registry.fill(HIST("hVtx3ProngY"), secondaryVertex3[1]);
+                  registry.fill(HIST("hVtx3ProngZ"), secondaryVertex3[2]);
```

Comment:
Do we need them given that we do not use the secondary vertex in the selection?

Reply:
No, they were already in place before so I did not delete them, but I could remove them

Reply:
I would remove them indeed if they are not necessary